### PR TITLE
Add stm32f072xb / stm32f072x8 HIC

### PR DESCRIPTION
**Pull request by mbrossard**
Based on #805 by @dotnfc (with additions from #947 from @zjli-2019)

### CODE REVIEW

File: source/hic_hal/stm32/stm32f072xx/IO_Config.h [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+/* SWD_CLK pin: PA2 */
+#define SWCLK_TCK_PORT      GPIOA
+#define SWCLK_TCK_PIN       GPIO_PIN_2
+
+/* SWD_NRST pin: PA1 */
+#define nRESET_PORT         GPIOA
+#define nRESET_PIN          GPIO_PIN_1
+
+/* SWD_SWO pin: PA3 */
```
[the remark this answers is not in the document]
Comment (mathias-arm) [reply source/hic_hal/stm32/stm32f072xx/IO_Config.h:88]: It seems to be a small mistake

**Comment by mbrossard** [comment]
@dotnfc / @zjli-2019: I noticed there are some unfinished work in `source/hic_hal/stm32/stm32f072xx/DAP_config.c` and `source/hic_hal/stm32/stm32f072xx/DAP_config.h` regarding `RESET_TARGET()` and `DAP_JTAG != 0`. It seems the TDI signal is not map to any GPIO.

**Comment by schuschu** [comment]
I am sorry if this is the wrong place (i.e., this qualifies as feature request) but from my understanding the lack of VTOR support in the CM0 means that the target does not support bootloader mode. 
However, at least the LPC11U35 HIC simply triggers the USB DFU (ROM bootloader) instead. Is this something that should be considered at the board/PCB level (e.g., tying BOOT0 to the target's reset pin) or could this be done in DAPLink by asserting the button (as done in the bootloader) and jumping to 0x1fffC804 to trigger the ROM bootloader (and thus USB DFU)?

**Comment by mbrossard** [comment]
> I am sorry if this is the wrong place (i.e., this qualifies as feature request) but from my understanding the lack of VTOR support in the CM0 means that the target does not support bootloader mode. However, at least the LPC11U35 HIC simply triggers the USB DFU (ROM bootloader) instead. Is this something that should be considered at the board/PCB level (e.g., tying BOOT0 to the target's reset pin) or could this be done in DAPLink by asserting the button (as done in the bootloader) and jumping to 0x1fffC804 to trigger the ROM bootloader (and thus USB DFU)?

When there is no bootloader (for lack of VTOR, but also more generally constraint on the amount of flash available) it is currently expected to be handled at the board/PCB level. Detecting a button press at startup to trigger ROM bootloader would certainly be something possible to add.

**Pull request merged**